Form validation

1Jeferson opened the pull request:
- Empty field
![campo-vazio](https://github.com/user-attachments/assets/e51763fa-7223-47c5-9202-27c29fa5f7e0)



- Password shorter than 7 characters
![senha-menor-7](https://github.com/user-attachments/assets/3e47044c-049e-4d9b-803c-4485968a9b01)



- Password that must have letters and numbers
![senha-precisa-letras-numeros](https://github.com/user-attachments/assets/68211ba0-08a4-424c-88ab-0cc15f1497b1)

1Jeferson committed:
feat: implement login page with Tailwind CSS and custom button component

1Jeferson committed:
feat: add tailwind-merge for button styling and update welcome message

1Jeferson committed:
refactor: remove unused CSS import from App component

1Jeferson committed:
feat: implement login form with validation and error messages

lucasfpds commented:
Um vídeo era melhor do que várias imagens não acha?

lucasfpds commented on src/components/message/index.tsx:
```diff
@@ -0,0 +1,11 @@
+  message?: string;
+}
+
+const ErrorMessage = ({ message }: ErrorMessageProps) => {
```
É interessante a criação da interface de forma separada quando vc exporta ela, ou tem muitas props para receber. Nesse caso, não está errado, mas seu código seria mais enxuto só com a tipagem do message não?

```suggestion
const ErrorMessage = (message?: string) => {
```

lucasfpds commented on src/components/message/index.tsx:
```diff
@@ -0,0 +1,11 @@
+  message?: string;
+}
+
+const ErrorMessage = ({ message }: ErrorMessageProps) => {
```
É interessante a criação da interface de forma separada quando vc exporta ela, ou tem muitas props para receber. Nesse caso, não está errado, mas seu código seria mais enxuto só com a tipagem do message não?

```suggestion
const ErrorMessage = (message?: string) => {
```

1Jeferson replied:
Um vídeo seria melhor mesmo, anotada a dica!!
E a alteração já foi feita em relação ao código, valeu!

1Jeferson committed:
refactor: simplify ErrorMessage component by removing props interface

1Jeferson committed:
refactor: update ErrorMessage component to use destructuring for props

racsandoval commented on src/pages/login/index.tsx:
```diff
@@ -0,0 +1,80 @@
+  const validateForm = () => {
+    const newErrors: { email?: string; password?: string } = {};
+    if (!email) {
+      newErrors.email = 'E-mail é obrigatório';
```
seria uma boa checar o formato do email tb

1Jeferson replied:
Checagem feita!

1Jeferson committed:
feat: email format check

lucasfpds commented on src/pages/login/index.tsx:
```diff
@@ -0,0 +1,86 @@
+
+    if (!email) {
+      newErrors.email = 'E-mail é obrigatório.';
+    } else if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)) {
```
Olha, é uma boa prática manter a facilidade de leitura do seu código. 
Esse seu regex nesse if não tá tão legal, não seria mais agradável algo como:

```suggestion
      const validateForm = () => {
    const newErrors: { email?: string; password?: string } = {};
    const isValidEmailFormat = /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
    
    if (!email) {
      newErrors.email = 'E-mail é obrigatório.';
    } else if (!isValidEmailFormat) {
```

Não é sempre só você que vai codar em um local, é importante deixar as coisas mais fáceis para os demais.

1Jeferson replied:
Alteração feita, valeu!

1Jeferson committed:
fix: correct email and password validation logic

- Ensure email format is valid
- Ensure password has at least 7 characters
- Ensure password contains at least one number and one letter

1Jeferson committed:
feat: add login validation with Zod and react-hook-form

racsandoval commented on package.json:
```diff
@@ -10,20 +10,26 @@
   "dependencies": {
-    "react": "^18.3.1",
-    "react-dom": "^18.3.1"
+    "@hookform/resolvers": "^3.10.0",
```
faltou fixar essa versão

racsandoval commented on src/schemas/index.ts:
```diff
@@ -0,0 +1,13 @@
+  email: z
+    .string()
+    .nonempty({ message: 'E-mail é obrigatório.' })
+    .regex(/^[^\s@]+@[^\s@]+\.[^\s@]+$/, { message: 'Por favor, insira um e-mail válido.' }),
```
o zod tem um validador de email, da uma olhada nele

1Jeferson replied:
Ajustado!

1Jeferson committed:
fix: adjust email validation to use zod.email method
